### PR TITLE
Show elixir 1.4.1 compiler msgs and unit test failures in Linter (fixes #8)

### LINKED ISSUE #8: Elixir 1.4.1
Noticing a number of errors, warnings and unit test failures are not parsed and sent to Linter for display. I suspect the compiler / exunit output format has changed over time. Do you accept pull requests?

### PR DESCRIPTION
**Pull request by brianlow**

### CODE REVIEW

File: lib/elixir.js [hunk 1]
```diff
@@ -18,29 +18,89 @@ export function provideBuilder() {
...
+      };
+
       return [ {
         name: 'Elixir: mix compile',
         exec: exec,
         args: [ 'compile' ],
         sh: false,
-        errorMatch: [ '\\*\\*\\s{1}\\((?<errorType>[a-zA-Z]+)\\)\\s{1}(?<file>[\\/0-9a-zA-Z\\\\._-]+).*\\s{1}starting at line\\s{1}(?<line>\\d+)',
-                      '\\*\\*\\s{1}\\((?<errorType>[a-zA-Z]+)\\)\\s{1}(?<file>[\\/0-9a-zA-Z\\\\._-]+):(?<line>\\d+)' ]
+        functionMatch: functionMatch
```
Comment (brianlow) [inline lib/elixir.js:89]: This change might introduce regressions but not sure how to replicate the compiler errors these were catching. I am thinking the important part of this PR is adding the specs so there is a way to handle new situations without breaking prior code.

File: package.json [hunk 2]
```diff
@@ -18,13 +18,17 @@
...
   },
   "keywords": [
     "build",
     "compile",
     "elixir",
     "productivity"
   ],
-  "main": "lib/elixir.js"
+  "main": "lib/elixir.js",
+  "scripts": { "test": "atom --test spec/*" }
```
Comment (brianlow) [inline package.json:33]: All tests pass. eslint is happy.

File: spec/examples/error/lib/error.ex [hunk 3]
```diff
@@ -0,0 +1,4 @@
+defmodule Error do
+	def go do
+	# missing end
+end
```
Comment (brianlow) [inline spec/examples/error/lib/error.ex:4]: The Elixir code that triggers a compiler error. One folder per compiler msg we are testing.

File: spec/examples/error/mix.log [hunk 4]
```diff
@@ -0,0 +1,6 @@
+Compiling 1 file (.ex)
+
+== Compilation error on file lib/error.ex ==
+** (TokenMissingError) lib/error.ex:5: missing terminator: end (for "do" starting at line 1)
+    (elixir) lib/kernel/parallel_compiler.ex:117: anonymous fn/4 in Kernel.ParallelCompiler.spawn_compilers/1
+
```
Comment (brianlow) [inline spec/examples/error/mix.log:6]: The .js specs just use these `mix.log` files. The specs don't require Elixir. The Elixir code is just included for reference.

File: spec/examples/run_mix.sh [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+rm -rf multiline_warning/_build/
+rm -rf unit_test/_build/
+rm -rf multiple_unit_tests/_build/
+
+docker run --rm --volume $PWD:/src --workdir /src/error                 elixir:1.4.1-slim mix compile > error/mix.log 2>&1
+docker run --rm --volume $PWD:/src --workdir /src/multiline_error       elixir:1.4.1-slim mix compile > multiline_error/mix.log 2>&1
+docker run --rm --volume $PWD:/src --workdir /src/warning               elixir:1.4.1-slim mix compile > warning/mix.log 2>&1
+docker run --rm --volume $PWD:/src --workdir /src/multiline_warning     elixir:1.4.1-slim mix compile > multiline_warning/mix.log 2>&1
+docker run --rm --volume $PWD:/src --workdir /src/unit_test             elixir:1.4.1-slim mix test    > unit_test/mix.log 2>&1
+docker run --rm --volume $PWD:/src --workdir /src/multiple_unit_tests   elixir:1.4.1-slim mix test    > multiple_unit_tests/mix.log 2>&1
```
Comment (brianlow) [inline spec/examples/run_mix.sh:17]: Not sure if this is useful. It documents how to generate the `mix.log` compiler output that the specs use.

File: .eslintrc [hunk 6]
```diff
@@ -1,4 +1,7 @@
 {
   "parser": "babel-eslint",
-  "extends": "atom-build"
+  "extends": "atom-build",
+  "rules": {
+    "no-cond-assign": [2, "except-parens"]
```
Comment (brianlow) [inline .eslintrc:5]: Lets you do assignments inside `if` statements as long as you surround with extra set of parens

File: package.json (hunk 2 above)
Comment (noseglid) [reply package.json:33]: This package has a `.travis.yml` but never really run or used.

I'm happy to merge this without travis support, but if you wanna add it, the instructions on how to do so can [be found here](https://github.com/atom/ci#travis-ci).

**Comment by noseglid** [comment]
This all looks really nice, and well done with specs! Please reply to my comment on how you wanna proceed.

I'd also be very happy if you would want to take maintenance resposibility over this package. I don't really use Elixir, and this package would benefit greatly if it was maintained by someone who does :)

**Comment by brianlow** [comment]
Will try to get travis working. Just learning Elixir right now, will be happy to maintain if I start using it full time.

**Comment by brianlow** [comment]
It worked!

**Comment by noseglid** [comment]
Very nice! Thank you very much for this excellent PR!

**Pull request merged**